qanta/buzzer/interface.py

```diff
@@ -0,0 +1,254 @@
+import csv
+import time
+import codecs
+import pickle
+import pandas as pd
+import itertools
+from functools import partial
+from functional import seq
+
+from qanta.util import constants as c
+from qanta.datasets.quiz_bowl import QuestionDatabase
+from qanta.guesser.abstract import AbstractGuesser
+from qanta import logging
+
+from util import *
+from performance import Prediction, Meta
+from multiprocessing import Pool
+
+log = logging.get(__name__)
+
+def process_vw_row(line):
+    line = line.split('|')
+    qid, sent, token = map(int, line[0].split('\'')[1].split('_'))
+    features = dict()
+    for section in line[1:-1]:
+        section = section.strip().split(' ')
+        section_name = section[0]
+        for feature in section[1:]:
+            feature_name, score = feature.split(':')
+            feature_name = section_name + '^' + feature_name
+            score = float(score)
+            features[feature_name] = score
+    guesser = line[-1].strip().split(' ')
+    guess = guesser[1]
+    scores = dict()
+    for feature in guesser[2:]:
+        feature_name, score = feature.split(':')
+        score = float(score)
+        scores[feature_name] = score
+    features['guess'] = (guess, scores['DAN_score'])
+    return qid, sent, token, guess, features, scores['DAN_score']
+
+def process_question(option2id, all_questions, qnum_q):
+    qnum, q = qnum_q
+    answer = format_guess(all_questions[qnum].page)
+
+    guess_vecs = []
+    results = []
+    prev_vec = [0 for _ in range(NUM_GUESSES)]
+    prev_dict = {}
+    for sent_token, group in q.groupby(['sentence', 'token'], sort=True):
+        group = group.sort_values('score', ascending=False)[:NUM_GUESSES]
+
+        # check if top guess is correct
+        top_guess = group.guess.tolist()[0]
+        results.append(int(top_guess == answer))
+
+        # get the current input vector
+        curr_vec = group.score.tolist()
+        curr_dict = {x.guess: x.score for x in group.itertuples()}
+        diff_vec, isnew_vec = [], []
+        for i, x in enumerate(group.itertuples()):
+            if x.guess not in prev_dict:
+                diff_vec.append(x.score)
+                isnew_vec.append(1)
+            else:
+                diff_vec.append(x.score - prev_dict[x.guess])
+                isnew_vec.append(0)
+        vec = curr_vec + prev_vec + diff_vec + isnew_vec
+        assert(len(vec) == 4 * NUM_GUESSES)
```

Comment:
Raise `ValueError` instead of `assert`, `assert`s can be turned off by the caller